### PR TITLE
Add splitlines

### PR DESCRIPTION
**Pull request by mpastell**
Adds `splitlines` as suggested in #19759

### CODE REVIEW

File: base/strings/util.jl [hunk 1]
```diff
@@ -297,6 +297,17 @@ end
...
+"""
+    splitlines(str::AbstractString; chomp = true)
+
+Return an array of strings by splitting the given string at line break characters
+(`'\\n'` or `"\\r\\n"`). When `chomp` is true (as it is by default),
+these trailing newline characters are removed from the
+line before it is returned. When `chomp` is false, they are returned as part of the
+line.
+"""
+splitlines(str::AbstractString; chomp = true) = readlines(IOBuffer(str), chomp = chomp)
```
Comment (KristofferC) [inline base/strings/util.jl]: Could you add a doctest here showing how it is used?

**Review by StefanKarpinski — APPROVED** [review]
LGTM, aside from a doctest example being nice (especially since folks have been working so hard at adding doctests to so many things).

**Comment by stevengj** [comment]
Or maybe just implement `readlines(s::AbstractString)` and `eachline(s::AbstractString)`  methods?

**Comment by StefanKarpinski** [comment]
```jl
julia> readlines("/usr/share/dict/words")
235886-element Array{String,1}:
 "A"
 "a"
 "aa"
 "aal"
 "aalii"
 "aam"
 "Aani"
 "aardvark"
 "aardwolf"
 ⋮
 "zymurgy"
 "Zyrenian"
 "Zyrian"
 "Zyryan"
 "zythem"
 "Zythia"
 "zythum"
 "Zyzomys"
 "Zyzzogeton"
```

**Comment by stevengj** [comment]
Oh, right, I forgot that you can pass the filename to `readlines` and `eachline`.

**Comment by stevengj** [comment]
I'm not sure calling `readlines` on an IOBuffer is the right implementation here, though.  The `split` function returns an array of substrings, so shouldn't `splitlines` do the same?

**Comment by stevengj** [comment]
The `split` function takes a duck-typed `splitter` argument that just has to work with `search`.   Maybe the right thing to do would be to define a new type, something like:
```julia
immutable LinebreakSplitter; chomp::Bool; end
function search(s::AbstractString, splitter::LinebreakSplitter, i::Integer)
       newline = search(s, '\n', i)
       splitter.chomp || return nextind(s, newline):newline
       prev = prevind(s, newline)
       return prev ≥ start(s) && s[prev] == '\r' ? prev:newline : newline:newline
end
splitlines(s::AbstractString; chomp::Bool=true) = split(s, LinebreakSplitter(chomp))
```
Not only does this re-use the `split` logic, but also this way the `LinebreakSplitter` could potentially be used with other functions (e.g. `replace`).

**Comment by mpastell** [comment]
Do we actually need the `chomp` option?

```julia
split(string, r"\n|\r\n")
```
would be a simple implementation that is compatible with `split`.

**Comment by StefanKarpinski** [comment]
I agree that `LinebreakSplitter` seems overly complicated. @mpastell's definition of `splitlines` or something equivalent to it seems sufficient to me.

**Comment by stevengj** [comment]
I'm not sure how important it is in practice, but Python's version has the equivalent of the `chomp=false` option.  I guess this would matter if you wanted to write the data back out while preserving the line endings.

**Comment by stevengj** [comment]
The `LinebreakSplitter` is only about 2x faster than the regex version in a simple test on `join([randstring(50)*(rand(Bool)?"\n":"\r\n") for i = 1:10^4])`, so I would only use it if we decide the `chomp` option is important.

**Comment by brenhinkeller** [comment]
So it looks like as of Julia 1.8.3 this is now provided by split:
```
julia> s = "this\nstring has several\nlines"
"this\nstring has several\nlines"

julia> split(s, '\n')
3-element Vector{SubString{String}}:
 "this"
 "string has several"
 "lines"
```
so I think we can close this, but if anyone disagrees (or wants to add an alias `splitlines(s) = split(s, `\n`)` feel free to reopen or make a new PR!

**Comment by mlhetland** [comment]
Does this mean `split` now handles errors with `\r` (which was the point of this issue and PR)? If not, maybe at least the issue should be kept open?

**Comment by brenhinkeller** [comment]
Ah I missed the point

**Comment by vtjnash** [comment]
Quite conflicted for such a small change. The implementation here is already mentioned now in the docstring for `eachline` though and seems trivial enough not to need this PR to add it
```
help?> eachline

  eachline(io::IO=stdin; keep::Bool=false)
  eachline(filename::AbstractString; keep::Bool=false)

...

  To iterate over each line of a String, eachline(IOBuffer(str)) can be used.
```

**Pull request closed without merging**